### PR TITLE
Remove obsolete config/olm files

### PR DESCRIPTION
**Pull request by StefanHauth**
## Description

Remove config/olm directory and clean out dependencies in other files

## How can this be tested?
Not at all.

## Checklist

- [n/a] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: config/olm/README.md [hunk 1]
```diff
@@ -1,84 +0,0 @@
-# How To Test OLM
```
Comment (0sewa0) [inline config/olm/README.md:1]: don't remove this

File: hack/build/bundle.sh [hunk 2]
```diff
@@ -1,75 +0,0 @@
-#!/bin/bash
```
Comment (0sewa0) [inline hack/build/bundle.sh:1]: don't remove this

File: hack/make/bundle.mk [hunk 3]
```diff
@@ -1,31 +0,0 @@
-# Current Operator version
```
Comment (0sewa0) [inline hack/make/bundle.mk:1]: don't remove this

File: hack/make/tests/olm.mk [hunk 4]
```diff
@@ -1,3 +0,0 @@
-## Installs a custom catalog source and adds a Dynatrace Operator subscription to test OLM installs
-test/olm:
```
Comment (0sewa0) [inline hack/make/tests/olm.mk:2]: would be interesting to test out if this still works

File: hack/setup_olm_catalog.sh [hunk 5]
```diff
@@ -1,69 +0,0 @@
-#!/bin/bash
```
Comment (0sewa0) [inline hack/setup_olm_catalog.sh:1]: would be interesting to test out if this still works

**Review by 0sewa0 — DISMISSED** [review]

**Review by andriisoldatenko — APPROVED** [review]
LGTM. Not sure how can I test it?

**Pull request merged**